File: src/main/java/edu/croc/analytics/dto/Order.java
package edu.croc.analytics.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public record Order (
        @JsonProperty("user_id")
        String userId,

        @JsonProperty("ordered_at")
        LocalDateTime orderedAt,

        @JsonProperty("status")
        Status status,

        @JsonProperty("total")
        BigDecimal total) {

    public enum Status {
        COMPLETED,
        CANCELED,
        CREATED,
        DELIVERY
    }
}

File: testTask/Order.cs
using Newtonsoft.Json;

namespace testTask{
    /// <summary>
    /// Вспомогательный класс для десериализации JSON-строки в список
    /// </summary>
    public class Order
    {
        
        [JsonProperty("user_id")]
        public required string UserId { get; set; }

        [JsonProperty("ordered_at")]
        public DateTime OrderedAt { get; set; }

        [JsonProperty("status")]
        public required string Status { get; set; }

        [JsonProperty("total")]
        public double Total { get; set; }
    }
}